app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java

```diff
@@ -181,14 +197,25 @@ public void displayScanningContent() {
   public void stopScanningContent() {
     networkText.setVisibility(View.GONE);
     permissionButton.setVisibility(View.GONE);
-    libraryList.removeFooterView(progressBar);
+    swipeRefreshLayout.setRefreshing(false);
     TestingUtils.unbindResource(LibraryFragment.class);
   }
 
   public void noNetworkConnection() {
     displayNoNetworkConnection();
   }
 
+  public void refreshFragment() {
```

Comment:
I'm not 100% happy with the refreshFragment method, but I couldn't find a better way to trigger a reload instead of waiting for the broadcast receiver to do it's job when the connection was available.

Reply:
This looks good to me.